gitbook-docs/market/user-guides/sign-up.md

```diff
@@ -1,24 +1,40 @@
 # Sign up
 
-Access to the market requires authentication. Please ensure you have a valid username/password. If you have not registered, please look at the instructions on how to do so [here](../front-end.md#new-user-signup) via the front end.
+Access to the market requires authentication.
+Make sure you have a valid username and password.
+If you have not registered, please look at the instructions on how to do so
+[via the web interface](../front-end.md#new-user-signup).
 
-Or, you can use the below command line in the `proof-market-toolchain` repository.
+Or, you can use the following command in the `proof-market-toolchain` repository.
 
 ## User
 
-Users can submit/retrieve orders on the proof market. You cannot generate proofs if you sign up as an ordinary user.
+Signup for an ordinary user through CLI looks like this:
 
-```
-python3 scripts/signup.py user -u <username> -p <password> -e <e-mail>
+```bash
+python3 scripts/signup.py user \
+    -u <username> \
+    -p <password> \
+    -e <e-mail>
 ```
 
-## Proof Producer
+If the signup is successful, this command creates `.user` and `.secret` files with your credentials
+in the `./scripts` directory.
 
-Proof Producers can submit/retrieve orders on the proof market and have the additional ability to submit/generate proofs.&#x20;
+All users can submit and retrieve orders on Proof Market, but only proof producers
+can generate and submit proofs.
 
-```
-python3 scripts/signup.py producer -u <username> -p <password> -e <e-mail>
-```
+## Proof producer
 
-If successful, this creates a _.user_ and a _.secret_ file in `scripts/`  the directory with your credentials.
+First, you should sign up or [sign in](../cmd-reference/user.md#signing-in) as a regular user,
+and then you can register yourself as a proof producer:
+
+```bash
+python3 scripts/signup.py producer \
+    -e <your Ethereum address> \
+    [-d <producer's description>] \
+    [-u <link to a website>] \
+    [-l <logo>]
```

Comment:
showing optional parameters? good, I like this syntax

Reply:
Yep. And know I think we should do it in all the other code examples, because now it's not clear which parameters are required and which are not.
What do you think?